pkg/utils/node/predicates.go

```diff
@@ -16,6 +16,7 @@ package node
 
 import (
 	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/utils/scheduling"
```

Comment:
I think this dependency array is a bit hairy. Should node rely on scheduling? What if scheduling wants to rely on node? It's not clear to me which way the arrow should go.

Reply:
It makes sense to have the `IsUnderutilized()` function here in the node utils folder. I think it does require knowing if a pod is ignored or not. I can see in the future us having pods be aware of what nodes they can schedule on, such as taints and tolerations. Maybe an idea is to compile the pods and nodes utils in the same folder?

Reply:
Probably worth an offline discussion but we should try to avoid 'utils' packages if we can